### PR TITLE
SQL-504: Tag triggered releases

### PR DESCRIPTION
**Pull request by nbagnard**
The build script set the project version at the beginning using getGitVersion(). This one is used in the manifest.
The generated jar names use getReleaseVersion() which has the abbreviated version to keep the names as they were before.

Examples :
- Building a "dirty" driver locally: 
getGitVersion => 1.0.3-54-g472accaae-dirty
getReleaseVersion => 1.0.3-54-SNAPSHOT
Driver name = mongodb-jdbc-1.0.3-SNAPSHOT.jar
Driver version = 1.0.3-54-g472accaae-dirty

- Building a tagged driver (expected behavior 😉):
getGitVersion => 1.1.1
getReleaseVersion => 1.1.1
Driver name = mongodb-jdbc-1.1.1.jar
Driver version = 1.1.1

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -100,6 +101,10 @@ configurations {
     integrationTestRuntimeOnly.extendsFrom runtimeOnly
     tdvtTestImplementation.extendsFrom implementation
     tdvtTestRuntimeOnly.extendsFrom runtimeOnly
+
+    // Use the release version instead of gitVersion for the produced jar
+    jar.archiveVersion = releaseVersion
+    shadowJar.archiveVersion = releaseVersion
```
Comment (bucaojit) [inline build.gradle:107]: I think it was brought up in the meeting, did we decide on whether we wanted to add more of the git version to the produced jar file to know which checkin it was built from?

Comment (nbagnard) [reply build.gradle:107]: From my notes, having the "full" version in the manifest was Ok and enough for us to be able to track which commit the snapshot was linked to. If I misunderstood, I can change releaseVersion to gitVersion first thing tomorrow.

Comment (bucaojit) [reply build.gradle:107]: Ah that's what it was.  Yes I see the manifest has the version with the git hash in it. 👍

**Review by bucaojit — APPROVED** [review]
Looks good. 
Built and tested it and the driver worked and has `Implementation-Version` set.

**Review by rychipman — APPROVED** [review]
LGTM! thank you for picking up this fix!

**Comment by nbagnard** [comment]
evergreen merge

**Pull request merged**